commit 2b6c6efbf555a0d7801764d5f160fcdc50da88db
Author: Andy Whitcroft <apw@canonical.com>
Date:   Fri Nov 27 17:38:30 2015 +0000

    UBUNTU: SAUCE: (no-up) add compat_uts_machine= kernel command line override
    
    We wish to use the arm64 buildds to build armhf binaries in 32bit chroots.
    To make this work we need uname to return armv7l machine type.  To achieve
    this add a kernel command line override for the 32bit machine type.
    Add compat_uts_machine=<type> to allow the LINUX32 personality to return
    that type for uname.
    
    Signed-off-by: Andy Whitcroft <apw@canonical.com>

diff --git a/kernel/sys.c b/kernel/sys.c
index 89d5be4..cbe75f6 100644
--- a/kernel/sys.c
+++ b/kernel/sys.c
@@ -1099,6 +1099,21 @@ out:
 DECLARE_RWSEM(uts_sem);
 
 #ifdef COMPAT_UTS_MACHINE
+static char compat_uts_machine[__OLD_UTS_LEN+1] = COMPAT_UTS_MACHINE;
+
+static int __init parse_compat_uts_machine(char *arg)
+{
+	strncpy(compat_uts_machine, arg, __OLD_UTS_LEN);
+	compat_uts_machine[__OLD_UTS_LEN] = 0;
+	return 0;
+}
+early_param("compat_uts_machine", parse_compat_uts_machine);
+
+#undef COMPAT_UTS_MACHINE
+#define COMPAT_UTS_MACHINE compat_uts_machine
+#endif
+
+#ifdef COMPAT_UTS_MACHINE
 #define override_architecture(name) \
 	(personality(current->personality) == PER_LINUX32 && \
 	 copy_to_user(name->machine, COMPAT_UTS_MACHINE, \
